core/domain/html_validation_service.py

```diff
@@ -831,3 +831,65 @@ def get_filename_with_dimensions(old_filename, exp_id):
     new_filename = regenerate_image_filename_using_dimensions(
         old_filename, height, width)
     return new_filename
+
+
+def add_image_id_and_remove_filepath_from_image_tag(
+        image_id_to_src_dict, html):
+    """Adds image_id-with-value in image tag.
+
+    Args:
+        html: str. HTML of content in which image tag present.
+        image_id_to_src_dict: dict. Dict contaning image id with source, image
+            ids that needs to be added in image tag.
+    Returns:
+        html: str. HTML contains image tag with image id and without filepath.
+    """
+    soup = bs4.BeautifulSoup(html.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        filepath = json.loads(unescape_html(image['filepath-with-value']))
+        for _id in image_id_to_src_dict:
+            if filepath == image_id_to_src_dict[_id]:
+                image['id-with-value'] = escape_html(json.dumps(_id))
+                del image_id_to_src_dict[_id]
+                break
+
+        del image['filepath-with-value']
+
+    return unicode(soup).replace('<br/>', '<br>')
+
+
+def get_image_src_from_html(html):
+    """Extracts source of each image present in the HTML.
+
+    Args:
+        html: str. the HTML in which the images are present.
+
+    Returns:
+        image_info_list: list. List contaning an image object.
+    """
+    image_src_list = []
+    soup = bs4.BeautifulSoup(html.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        filepath = json.loads(unescape_html(image['filepath-with-value']))
+
+        image_src_list.append(filepath)
+
+    return image_src_list
+
+
+def get_image_ids_from_image_tag(html):
+    """Returns a list of image_ids present in the html.
+
+    Args:
+        html: str. the HTML from which image_ids are extracted.
+
+    Returns:
+        image_ids. List of ids, of all images present in an exploration.
```

Comment:
```suggestion
        image_ids. List of all IDs of images present in the HTML.
```
I don't think this needs to be for exploration only. Mention images in the HTML instead.

Reply:
done.